test/test_phyloflags.py

```diff
@@ -0,0 +1,347 @@
+"""Some basic testing and demo code for the bito module.
+
+If you want to see the results of the print statements, use `pytest -s`.
+"""
+
+import json
+import pprint
+import pytest
+import numpy as np
+import bito
+import bito.beagle_flags as beagle_flags
+import bito.phylo_flags as flags
+import bito.phylo_model_mapkeys as model_keys
+import bito.phylo_gradient_mapkeys as gradient_keys
+import sys
+
+# DEMO
+
+
+def gradients_with_flags_demo():
+    inst = create_instance()
+    initialize_model_parameters(inst)
+
+    # Request and calculate gradients for RATIOS_ROOT_HEIGHT,
+    # SUBSTITUTION_MODEL_FREQUENCIES, SUBSTITUTION_MODEL_RATES
+    bito_grad = inst.phylo_gradients(
+        # explicit flags: For flags that don't have associated values, can just be a list.
+        [
+            flags.RATIOS_ROOT_HEIGHT,
+            flags.SUBSTITUTION_MODEL,
+        ],
+        # run_with_default_flags:
+        # (1) If set to true, all fields of phylo_model_block_map are populated unless overriden by explicit flag.
+        # (2) If set to false, no fields of phylo_model_block_map are populated unless overriden by explicit flag.
+        False,
+    )[0]
+
+    print("bito_grad_keys: ", bito_grad.gradient.keys())
+    gtr_rates_grad = np.array(
+        bito_grad.gradient[gradient_keys.SUBSTITUTION_MODEL_RATES]
+    )
+    gtr_freqs_grad = np.array(
+        bito_grad.gradient[gradient_keys.SUBSTITUTION_MODEL_FREQUENCIES]
+    )
+    ratios_root_height = np.array(bito_grad.gradient[gradient_keys.RATIOS_ROOT_HEIGHT])
+
+    print("GTR rates gradient: \n", gtr_rates_grad)
+    print("GTR freqs gradient: \n", gtr_freqs_grad)
+    print("root height gradient: \n", ratios_root_height)
+
+    # above works if only boolean flags are used, otherwise:
+    bito_grad = inst.phylo_gradients(
+        # explicit flags: For SET flags that require value, use ordered tuples. Non-SET flags just take boolean.
+        [
+            (flags.SET_GRADIENT_DELTA, 5.0),
```

Comment:
It seems to me that with the templating we can either run with this flavor of flag, or with the flavor of flag in the previous example, but not mix and match. Is that correct?

Reply:
Yes, that is correct.